src/main/java/com/ita/if103java/ims/dao/impl/AccountTypeDaoImpl.java

```diff
@@ -0,0 +1,158 @@
+            throw crudException(e.getMessage(), "get", "name = " + name);
+        }
+    }
+
+    @Override
+    public AccountType update(AccountType accountType) {
+        int status;
+        try {
+            status = jdbcTemplate.update(
+                Queries.SQL_UPDATE_ACCOUNT_TYPE,
+                accountType.getName(),
+                accountType.getPrice(),
+                accountType.getMaxWarehouses(),
+                accountType.getMaxWarehouseDepth(),
+                accountType.getMaxUsers(),
+                accountType.getMaxSuppliers(),
+                accountType.getMaxClients());
+
+        } catch (DataAccessException e) {
+            throw crudException(e.getMessage(), "update", "id = " + accountType.getId());
+        }
+        if (status == 0)
+            throw accountTypeEntityNotFoundException("Update account type exception", "id = " + accountType.getId());
+
+        return accountType;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        int status;
+        try {
+            status = jdbcTemplate.update(Queries.SQL_SET_ACTIVE_STATUS_ACCOUNT_TYPE, false, id);
+
+        } catch (DataAccessException e) {
+            throw crudException(e.getMessage(), "delete", "id = " + id);
+        }
+        if (status == 0)
+            throw accountTypeEntityNotFoundException("Delete account type exception", "id = " + id);
+
+        return true;
+    }
+
+    private PreparedStatement createStatement(AccountType accountType, Connection connection) throws SQLException {
+        PreparedStatement preparedStatement = connection.prepareStatement(Queries.SQL_CREATE_ACCOUNT_TYPE, Statement.RETURN_GENERATED_KEYS);
+
+        int i = 0;
+        preparedStatement.setString(++i, accountType.getName());
+        preparedStatement.setDouble(++i, accountType.getPrice());
+        preparedStatement.setInt(++i, accountType.getMaxWarehouses());
+        preparedStatement.setInt(++i, accountType.getMaxWarehouseDepth());
+        preparedStatement.setInt(++i, accountType.getMaxUsers());
+        preparedStatement.setInt(++i, accountType.getMaxSuppliers());
+        preparedStatement.setInt(++i, accountType.getMaxClients());
+        preparedStatement.setBoolean(++i, accountType.isActive());
+
+        return preparedStatement;
+    }
+
+    private EntityNotFoundException accountTypeEntityNotFoundException(String message, String attribute) {
+        EntityNotFoundException exception = new EntityNotFoundException(message);
+        logger.error("EntityNotFoundException exception. Account type is not found ({}). Message: {}", attribute, message);
+        return exception;
+    }
+
+    private CRUDException crudException(String message, String operation, String attribute) {
+        CRUDException exception = new CRUDException(message);
+        logger.error("CRUDException exception. Operation:({}) account type ({}) exception. Message: {}", operation, attribute, message);
+        return exception;
+    }
```

Comment:
Please sit together with the team and think how to move this similar methods to some abstract class and generalize them in order they can be accessed by every DAO